src/dataflow/src/render/reduce.rs

```diff
@@ -731,16 +787,34 @@ where
 }
 
 /// Build the dataflow to compute the set of distinct keys.
-fn build_distinct<G>(collection: Collection<G, (Row, Row)>) -> Arrangement<G, Row>
+fn build_distinct<G, T>(
+    collection: Collection<G, (Row, Row)>,
+    err_input: Collection<G, DataflowError>,
+) -> CollectionBundle<G, Row, T>
```

Comment:
We can discuss further, but I think we probably want to keep both options around and allow the planner to distinguish (from prior experiences where different users want different things, and its easier to flip flop a plan around than the rendering behavior).

Reply:
Both variants are around now and can be selected through the plan.